### ISSUE TITLE
`Enhancement` Keep the screen on while sensor is active

### ISSUE THREAD

**Issue by nijel** [Blackymas/NSPanel_HA_Blueprint#1953]
### Enhancement Summary

System settings - Wake-up sensors should keep the screen on as long as sensor is active.

### Detailed Description

I have a motion sensor in a room where NSPanel is placed, and I want NSPanel to be on as long as there is motion going on. Currently, it only turns on when motion is detected for the first time, and then the usual dimming and sleep timeouts apply.

### Additional Context

Thanks for this blueprint, it works great!

**Comment by sgurgul** [comment]
@nijel May I ask how do you wake up the screen ? What action do you do on nspanel device to turn it's (sleeping) screen on ?

**Comment by nijel** [comment]
The motion sensor is selected in blueprint (System settings - Wake-up sensors).

**Comment by sgurgul** [comment]
Yes, useful, indeed. And the improvement proposed in this ticket would be very welcome as well :)
Thank you for answer.

**Comment by snizzleorg** [comment]
It would be great if this also could be connected to the state of the lights in the room. 
lights on => no switching off display, lights off => light goes off after delay.

**Comment by arielwulkan** [comment]
Also would be great to have API command to keep screen on regardless of sensor

**Comment by mcsdodo** [comment]
+1 on this. I'm using one panel as a "dashboard" under my PC monitor. Would be nice if it could be always-on as long as I'm in the room (this I have handled by ble presence).

**Comment by minimag450** [comment]
In the meantime, is there a solution so that the panel stays on as long as the motion detector detects movement and switches off when the motion detector becomes free? I have an Aqara FP2 in use. 
Greetings, Andreas

**Comment by mcsdodo** [comment]
I have an automation to do that. Every interval X I toggle an helper _input_boolean_ (if presence is detected). Then I have a template _binary_sensor_ which I then use in _wake_up_sensors_ configuration of this blueprint. Works okay.

**Comment by edwardtfn** [comment]
Look, we don't have anything specific for this, but this shouldn't be that hard to implement, however, I'm totally out of time lately, so I'm not sure if I will be able to invest time on this in the coming weeks.

Anyways, it's important you understand how this works... We have 3 timers, one for falling back to the home page, one for dimming and one for sleeping. Any of those timers are set by a number entity on your panel's page (Settings > Devices & services > ESPHome) and you can control those in automations.
When set to 0, the respective function will be disabled, which means, if you set the timeout sleep to 0, it will never sleep.

In parallel of this automatic timers, you also have a service to wake-up the panel. There's no service to force sleep, but calling the command service with the command `page screensaver` will have the same result.

So, you can play with that in your automations. If you set your timeout sleep to 0, the panel won't be managing the sleep anymore, so you can do in your external automation.
You can, as an example, create an automation to call the `wake_up` action when your sensor detects motion and another to call `page screensaver` when your sensor doesn't detect motion anymore. You can even set some long time for sleep in your panel's settings just to work as a fallback engine in case your panel is out of wifi or your automation fails.

As an easier alternative, you can find a ["set display as a light" in our customization page](https://github.com/Blackymas/NSPanel_HA_Blueprint/blob/main/docs/customization.md#set-display-as-a-light). That will create a light in your Home Assistant which is assigned to your display. If you turn-off the light, the display will wake-up, if you turn-off, it will go to sleep. And you can set the brightness.
With that, you can use any of the available blueprints to turn lights base on motion, and that will get exactly what you want.

Please take a look at those options and let me know your thoughts.
I still wanna look for a more elegant solution in the future, but it will be nice if you can use a workaround in the meantime and get to your target earlier. 😉 

![Image](https://github.com/user-attachments/assets/da8e49d2-1bf3-4cab-b6e6-67a6eb974189)

**Comment by minimag450** [comment]
Hi Edward, thank you for the detailed answer. I don't give every Rsg time to deal with it either. So please don't worry about it.
I will look at the suggested options and then give you an answer. The easiest option I see is to switch the panel on or off as a light.
Greetings, Andreas

**Comment by minimag450** [comment]
I had a look at the customisation page ( set display as a light). Do I understand it correctly? I just need to paste the code into my ESPHome configuration from the panel and then install the complete code on the panel?

Best regards, Andreas

**Comment by minimag450** [comment]
Is there another solution now? I can't get it to work with the suggested solutions.
